universal-application-tool-0.0.1/app/views/AzureFileUploadViewStrategy.java

```diff
@@ -0,0 +1,126 @@
+package views;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.each;
+import static j2html.TagCreator.footer;
+import static j2html.TagCreator.form;
+import static j2html.TagCreator.input;
+
+import controllers.applicant.routes;
+import j2html.tags.ContainerTag;
+import j2html.tags.Tag;
+import java.util.Optional;
+import javax.inject.Inject;
+import services.applicant.question.FileUploadQuestion;
+import services.cloud.FileNameFormatter;
+import services.cloud.StorageUploadRequest;
+import services.cloud.azure.BlobStorageUploadRequest;
+import views.applicant.ApplicantProgramBlockEditView.Params;
+import views.questiontypes.ApplicantQuestionRendererFactory;
+import views.questiontypes.ApplicantQuestionRendererParams;
+import views.style.ApplicantStyles;
+import views.style.Styles;
+
+public class AzureFileUploadViewStrategy extends FileUploadViewStrategy {
+
+  private final ViewUtils viewUtils;
+
+  @Inject
+  AzureFileUploadViewStrategy(ViewUtils viewUtils) {
+    this.viewUtils = checkNotNull(viewUtils);
+  }
+
+  @Override
+  public ContainerTag signedFileUploadFields(
+      ApplicantQuestionRendererParams params, FileUploadQuestion fileUploadQuestion) {
+    StorageUploadRequest storageUploadRequest = params.signedFileUploadRequest().get();
+
+    BlobStorageUploadRequest request = castStorageRequest(storageUploadRequest);
+
+    Optional<String> uploaded =
+        fileUploadQuestion.getFilename().map(f -> String.format("File uploaded: %s", f));
+
+    ContainerTag formTag = form();
+    return formTag
+        .with(div().withText(uploaded.orElse("There is no file uploaded")))
```

Comment:
suggestion (blocking): extract this UI text to a message in the i18n system https://github.com/seattle-uat/civiform/wiki/Internationalization-(i18n)

Reply:
Sorry, I realize I left this from debugging. The AWS code hides the string completely if there is no uploaded file, so I went ahead and changed my code for the sake of consistency.